### PR TITLE
Fix 33: Array to string conversion when putting a Rating Block

### PR DESCRIPTION
**Pull request by leal-k**
This is a proposed fix for the bug [#33](https://github.com/concrete5cojp/c5j_ratings/issues/33). 2 files are modified:
`src/Traits/RatingTrait.php` - Fixed the `$action` variable, making sure it contains a string and not an array, treating correctly the data it contains.
`src/Page/PageList.php` - Added 2 lines in the SQL query to `GROUP BY` correctly with `r.bID` and `p.cDisplayOrder`.

### CODE REVIEW

File: src/Traits/RatingTrait.php [hunk 1]
```diff
@@ -60,8 +60,10 @@ public function generate($action = '', $time = null, $uID = 0): string
             $time = time();
         }
         $config = $app->make('config/database');
-
-        return $time . ':' . md5($time . ':' . $uID . ':' . $action . ':' . $config->get('concrete.security.token.validation'));
+        $actionArray = array_map('strval', $action); //makes sure that $action is a string
+        $actionString = implode(':', $actionArray); //properly dealing with the array to string
+        $actionString = str_replace(' ', '_', $actionString); //treating all spaces in the string
```
Comment (biplobice) [inline src/Traits/RatingTrait.php:65]: @leal-k 
Can you please tell me under which circumstances the `$action` variable becomes an array? It should always be a string. If not, please correct the location from which it's being called.

File: src/Page/PageList.php [hunk 2]
```diff
@@ -18,6 +18,8 @@ public function __construct()
         $this->query->addSelect('SUM(r.ratedValue) AS ratings');
         $this->query->addSelect('r.bID');
         $this->query->addGroupBy('p.cID');
+        $this->query->addGroupBy('r.bID');
+        $this->query->addGroupBy('p.cDisplayOrder');
```
Comment (biplobice) [inline src/Page/PageList.php:22]: @leal-k 
If you're trying to fix any other issue, please make a separate PR.

File: src/Traits/RatingTrait.php (hunk 1 above)
Comment (leal-k) [reply src/Traits/RatingTrait.php:65]: When I was tracing the issue down to check why the array to string conversion was happening, I debugged all of the `return` statement just under this section of code, I found out that `$action` was returning an array with the name of the fields and the corresponding values after, but it should always be a string, as the function implied. After the treatment in the code and the addition of the SQL query conditions the bug stopped, both were needed. After that the issue with the database appeared, as I said in issue #35 . But the fix for that #35 issue, and consequently this issue was just the query needed the update, as I told on our conversation. If needed, I can replicate the issue and show the array that is being passed. Basically, the fix #36 resolved both issues, that was just the SQL query that needed to be updated.
